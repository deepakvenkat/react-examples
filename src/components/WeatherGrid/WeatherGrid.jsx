import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import WeatherCard from '../WeatherCard';

const styles = {};

const WeatherGrid = (({ days }) => {
  const weatherCardsList = days.map((day) => (
    <WeatherCard
      day={day.day}
      high={day.high}
      low={day.low}
      weather={day.weather}
      key={day.day}
    />
  ));
  return (
    <div>
      {weatherCardsList}
    </div>
  );
});

WeatherGrid.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  // TODO: Extract `day` to a common proptype
  days: PropTypes.arrayOf(PropTypes.shape({
    day: PropTypes.string.isRequired,
    high: PropTypes.string.isRequired,
    low: PropTypes.string.isRequired,
    weather: PropTypes.string.isRequired,
  })).isRequired,
};

export default withStyles(styles)(WeatherGrid);
