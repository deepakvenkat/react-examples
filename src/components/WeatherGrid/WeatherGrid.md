## WeatherGrid

```jsx
const days = [{
  weather: 'sunny',
  day: 'Sat',
  low: '65',
  high: '72',
}, {
  weather: 'cloudy',
  day: 'Sun',
  low: '60',
  high: '68'
}, {
  weather: 'overcast',
  day: 'Mon',
  low: '62',
  high: '70'
}];

<WeatherGrid days={days} />
```
