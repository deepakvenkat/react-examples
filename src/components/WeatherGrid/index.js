import WeatherGrid from './WeatherGrid.jsx';

export default WeatherGrid;
