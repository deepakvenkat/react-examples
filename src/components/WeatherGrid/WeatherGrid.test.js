import React from 'react';
import { render } from '@testing-library/react';
import WeatherGrid from './WeatherGrid';

test('renders the WeatherGrid', () => {
  const days = [{
    weather: 'sunny',
    day: 'Sat',
    low: '65',
    high: '72',
  }, {
    weather: 'cloudy',
    day: 'Sun',
    low: '60',
    high: '68'
  }, {
    weather: 'overcast',
    day: 'Mon',
    low: '62',
    high: '70'
  }];
  const { asFragment } = render(
    <WeatherGrid days={days} />
  );
  expect(asFragment()).toMatchSnapshot();
});
