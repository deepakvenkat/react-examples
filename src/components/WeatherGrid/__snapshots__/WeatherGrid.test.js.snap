// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the WeatherGrid 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root WeatherCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Sat
        </p>
        <div
          class="MuiAvatar-root MuiAvatar-square WeatherIcon-avatar-4 MuiAvatar-colorDefault"
        >
          <div
            class="WeatherIcon-first-3 WeatherIcon-first-5"
          />
        </div>
        <br />
        <p
          class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1"
        >
          72/
        </p>
        <p
          class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          65
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root WeatherCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Sun
        </p>
        <div
          class="MuiAvatar-root MuiAvatar-square WeatherIcon-avatar-4 MuiAvatar-colorDefault"
        >
          <div
            class="WeatherIcon-first-3 WeatherIcon-first-6"
          />
        </div>
        <br />
        <p
          class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1"
        >
          68/
        </p>
        <p
          class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          60
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root WeatherCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Mon
        </p>
        <div
          class="MuiAvatar-root MuiAvatar-square WeatherIcon-avatar-4 MuiAvatar-colorDefault"
        >
          <div
            class="WeatherIcon-first-3 WeatherIcon-first-7"
          />
        </div>
        <br />
        <p
          class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1"
        >
          70/
        </p>
        <p
          class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          62
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
